File: class_object_method.py

class Animal:
    name = ""

    def __init__(self, name):
        self.name = name
        print("This is constractor!")
    
    def eat(self):
        print(f"{self.name} can eat!")

    def sleep(self):
        print(f"{self.name} can sleep!")


cat = Animal("Kitty")
print(cat.name)
cat.eat()
cat.sleep()

File: class_object_method.cpp
#include<bits/stdc++.h>
using namespace std;

class Animal{
    public:
        string name;

        Animal(string name){
            this->name = name;

            cout<<"This is Constractor!"<<endl;
        }
        // insided method
        void eat(){
            cout<<name<<" can Eat"<<endl;
        }
        void sleep();
};
// outsided method
void Animal::sleep(){
    cout<<name<<" can sleep"<<endl;
}

int main(){

    Animal cat("Kitty");
    cat.eat();
    cat.sleep();

    return 0;
}